io/subsystem/src/main/java/org/wildfly/extension/io/BufferPoolResourceDefinition.java

```diff
@@ -55,66 +51,66 @@
     static final RuntimeCapability<Void> IO_POOL_RUNTIME_CAPABILITY =
             RuntimeCapability.Builder.of(IOServices.BUFFER_POOL_CAPABILITY_NAME, true, Pool.class).build();
 
-    private static final int defaultBufferSize;
-    private static final int defaultBuffersPerRegion;
-    private static final boolean defaultDirectBuffers;
+    static final int defaultBufferSize;
+    static final int defaultBuffersPerRegion;
+    static final boolean defaultDirectBuffers;
+
+    static final ModelNode DEFAULT_BUFFER_SIZE;
+    static final ModelNode DEFAULT_BUFFERS_PER_REGION;
+    static final ModelNode DEFAULT_DIRECT_BUFFERS;
 
     static {
         long maxMemory = Runtime.getRuntime().maxMemory();
-        //smaller than 64mb of ram we use 512b buffers
+        // smaller than 64mb of ram we use 512b buffers
         if (maxMemory < 64 * 1024 * 1024) {
-            //use 512b buffers
+            // use 512b buffers
             defaultDirectBuffers = false;
             defaultBufferSize = 512;
             defaultBuffersPerRegion = 10;
         } else if (maxMemory < 128 * 1024 * 1024) {
-            //use 1k buffers
+            // use 1k buffers
             defaultDirectBuffers = true;
             defaultBufferSize = 1024;
             defaultBuffersPerRegion = 10;
         } else {
-            //use 16k buffers for best performance
-            //as 16k is generally the max amount of data that can be sent in a single write() call
+            // use 16k buffers for best performance
+            // as 16k is generally the max amount of data that can be sent in a single write() call
             defaultDirectBuffers = true;
             defaultBufferSize = 1024 * 16;
             defaultBuffersPerRegion = 20;
         }
+        DEFAULT_BUFFER_SIZE = new ModelNode(defaultBufferSize);
+        DEFAULT_BUFFERS_PER_REGION = new ModelNode(defaultBuffersPerRegion);
+        DEFAULT_DIRECT_BUFFERS = new ModelNode(defaultDirectBuffers);
     }
 
     static final SimpleAttributeDefinition BUFFER_SIZE = new SimpleAttributeDefinitionBuilder(Constants.BUFFER_SIZE, ModelType.INT, true)
             .setFlags(AttributeAccess.Flag.RESTART_ALL_SERVICES)
             .setAllowExpression(true)
             .setValidator(new IntRangeValidator(1, true, true))
+            .setDefaultValue(DEFAULT_BUFFER_SIZE)
```

Comment:
You can not set default values for these attributes as they are derived from `Runtime.getRuntime().maxMemory()`.
Their actual values will depend on the VM they are running and will not be provide the same "default" across all servers in domain mode.